openapi/src/codegen.rs

```diff
@@ -1234,7 +1234,11 @@ pub fn gen_field_rust_type<T: Borrow<Schema>>(
     }
     if field_name == "metadata" {
         state.use_params.insert("Metadata");
-        return "Metadata".into();
+        return if !required || is_nullable {
```

Comment:
@arlyon after doing a quick search, this seemed the likely culprit, but feel free to do something else if this isn't the way to fix it.

Reply:
I have a PR that also applies to EventData also, closing in favour of that one :)